crates/storage/provider/src/providers/state/historical.rs

```diff
@@ -41,7 +41,7 @@ use std::fmt::Debug;
 /// - [`tables::AccountChangeSets`]
 /// - [`tables::StorageChangeSets`]
 #[derive(Debug)]
-pub struct HistoricalStateProviderRef<'b, TX: DbTx> {
+pub struct HistoricalStateProviderRef<'b, TX: DbTx, SC: StateCommitment> {
```

Comment:
this additional generic looks weird to me, because we only use it to capture the type